Python-packages/covidcast-py/covidcast/covidcast.py

```diff
@@ -324,6 +328,28 @@ def aggregate_signals(signals: list, dt: list = None, join_type: str = "outer")
     return joined_df
 
 
+def _parse_datetime_weeks(date_str: str,
+                          format: str="%Y%m%d") -> Union[pd.Timestamp]:  # annotating nan errors
+    """Convert a date or epiweeks string into timestamp objects.
+
+    Datetimes (length 8) are converted to their corresponding date, while epiweeks (length 6)
+    are converted to the date of the start of the week. Returns nan otherwise
+
+    Epiweeks use the CDC format.
+
+    :param date_str: String of date or epiweek
+    :param format: String of the date format to parse.
+    :returns: Timestamp
+    """
+    if len(date_str) == 8:
```

Comment:
Do you want to do this by using the `time_type` column instead of length? I don't know if we'll ever have two time_types with identical-length formatted dates, but it seems cleaner to use `time_type` anyway

Reply:
I'm going to go ahead and insist on switching to `time_type`

Reply:
ah, didn't realize that was a thing, good idea.

Reply:
a bit messy of a .apply statement since we need both columns, but implemented now